### PR TITLE
Restore support for Node.js<v6.13.0

### PR DESCRIPTION
**Pull request by AndreyBelym**
Unfortunately I can't add an older Node config to the TravisCI env, because `eslint` requires `node>=6.14.0` (see the tests for the first commit)

### CODE REVIEW

File: async.js [hunk 1]
```diff
@@ -24,32 +23,25 @@ var url = require('./url')
...
-          resolve(id)
-        }
-      })
-    })
+
+  if (!callback) {
+    return randomPromise
   }
+
+  randomPromise
```
Comment (ai) [inline async.js]: It will not work :D. `callback` API was created to support environments without `Promise`.

File: random.js [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+  return crypto.randomFillSync(Buffer.allocUnsafe(bytes))
+}
+
+module.exports.async = function (bytes) {
+  return new Promise(function (resolve, reject) {
+    function callback (error, buffer) {
+      error ? reject(error) : resolve(buffer)
+    }
+
+    if (crypto.randomFill) {
```
Comment (ai) [inline random.js]: If will be better to:

```js
if (crypto.randomFill) {
  module.exports.async = …
} else {
  module.exports.async = …
}
```

Will reduce extra check in ``random`.

Comment (ai) [reply random.js]: Yeap, I know that it is harder to test it :D

File: async.js (hunk 1 above)
Comment (AndreyBelym) [reply async.js]: Ok, I've got it. So I can't use Promise in `random.js` too :(

**Comment by AndreyBelym** [comment]
FPR

**Comment by ai** [comment]
Released in 1.2.5. Thanks for the help.

**Pull request merged**